process/modules/special_transport_routes.py

```diff
@@ -0,0 +1,201 @@
+            "toiminnal2",
+            "muuntaja2",
+            "tuleva2",
+            "varareitti2",
+            "muokattu",
+            "datanomistaja",
+            "paivitetty_tietopalveluun",
+            "ylre_class",
+            "kadun_nimi",
+            "gml_id",
+            "id",
+        ]
+
+        self._lines = gpd.read_file(file_name)
+        self._lines["vaylatyyp2"] = self._lines["vaylatyyp2"].str.replace("\n", "")
+
+        # Drop one invalid data line
+        self._lines = self._lines.drop(self._lines[(self._lines["id"] == 1500) | ((self._lines["nimi_1"] == "Metsälä-Etelä-Oulunkylä") & (self._lines["vaylatyyp2"] == "Kevyen liikenteen väylä"))].index)
+
+        self._orig = self._lines
+
+    def _keep_rows_base_on_hierarchy_list(self, types, shapes: gpd.GeoDataFrame) -> gpd.GeoDataFrame:
+        retval = shapes.copy()
+        mask = retval["hierarkia"].isin(types)
+        retval = retval[mask]
+        return retval
+
+    def _drop_not_used_classes_base_on_main_and_sub_types(self, main_and_sub_types, shapes: gpd.GeoDataFrame) -> gpd.GeoDataFrame:
+        retval = shapes.copy()
+        for main_type, sub_types in main_and_sub_types.items():
+            for sub_type in sub_types:
+                retval.drop(
+                    retval[
+                        (retval["paatyyppi"] == main_type)
+                        & (retval["alatyyppi"] == sub_type)
+                    ].index,
+                    axis=0,
+                    inplace=True,
+                )
+
+        return retval
+
+    def _drop_unnecessary_columns(self, columns_to_drop: list[str], shapes: gpd.GeoDataFrame) -> gpd.GeoDataFrame:
+        retval = shapes.copy()
+        retval.drop(columns_to_drop, axis=1, inplace=True)
+
+        return retval
+
+    def _buffering(self, lines: gpd.GeoDataFrame) -> gpd.GeoDataFrame:
+        # Buffer lines
+        target_infra_polys = lines.copy()
+        retval = target_infra_polys[0:0]
+        for buffer_class, buffer_value in self._buffers.items():
+            buffered_items = target_infra_polys.loc[target_infra_polys["vaylatyyp2"].isin(self._buffer_class_vaylatyp2_values[buffer_class])].copy()
+            buffered_items["geometry"] = buffered_items.buffer(buffer_value)
+            retval = gpd.GeoDataFrame(pd.concat([retval, buffered_items], ignore_index=True))
+
+        return retval
+
+    def _check_and_set_ylre_classes_id(self, lines: gpd.GeoDataFrame) -> gpd.GeoDataFrame:
+        attrs = ["ylre_class", "kadun_nimi"]
+        ylre_katualueet_dissolved = self._ylre_katualueet.dissolve(by=attrs)
+        ylre_katualueet_dissolved["geometry"] = ylre_katualueet_dissolved.buffer(15)
+        joined_result = gpd.sjoin(lines, ylre_katualueet_dissolved, predicate='within')
+
+        retval = lines.merge(joined_result[['gml_id','ylre_class','kadun_nimi']], how='left', left_on='gml_id', right_on='gml_id')
+
+        return retval
+
+    def _makeValid(self, data, geom_column):
```

Comment:
Can be static? Also, function names should be lowercase.

Reply:
Changed function name. Function is used in process.